### DISCUSSION TITLE
Why `valtio`?

### DISCUSSION THREAD

**Discussion by with-heart** [chakra-ui/zag#58] [Q&A] [answered]
Hi!

As the title suggests, just wondering why you decided to use `valtio` @segunadebayo! Would love to hear the benefits this gave you with `zag`!

I haven't spent much time playing with it yet but it seems like an interesting library. Convince me to use it more? 😁

**Comment by segunadebayo** [comment, marked as answer]
Hey @with-heart,

Valtio provided the base-level primitives we need for the machine state management.

- Fine-grained reactivity for subscriptions (using Proxy).
- Ability to simply mutate the `context` rather than wrapping things in `assign(...)`.
- Support for computed state properties like `proxyWithComputed`, `derive`, etc.
- When used with react, you get a `useSnapshot` hook that automatically optimizes re-renders based on property access.

These are some of the benefits we get from `valtio`. I think we're pretty flexible with the base library as long as it's able to support these features as well.

**Reply by with-heart** [reply]
Thank you for the detailed explanation!
